packages/omgidl-serialization/src/DeserializationInfoCache.ts

```diff
@@ -209,11 +238,116 @@ export class DeserializationInfoCache {
       name,
       type,
       typeDeserInfo: fieldDeserInfo,
+      isComplex: false,
       isArray,
       arrayLengths,
       definitionId: getDefinitionId(definition),
+      isOptional: isOptional(definition),
     };
   }
+
+  /** Returns default value for given FieldDeserializationInfo.
+   * If defaultValue is not defined on FieldDeserializationInfo, it will be calculated and set.
+   */
+  public getFieldDefault(deserInfo: FieldDeserializationInfo): unknown {
+    if (deserInfo.defaultValue != undefined) {
+      return deserInfo.defaultValue;
+    }
+
+    return this.#updateFieldDeserInfoWithDefaultValue(deserInfo);
```

Comment:
Not sure exactly what this function does, but I don't think you need it. You should be able to just call `getFieldDefault` recursively if required. Maybe something like this:

```ts
  public getFieldDefault(deserInfo: FieldDeserializationInfo): unknown {
    if (deserInfo.defaultValue != undefined) {
      return deserInfo.defaultValue;
    }

    if (deserInfo.typeDeserInfo.type === "primitive") {
      deserInfo.defaultValue = PRIMITIVE_DEFAULT_VALUE_GETTERS.get(deserInfo.type);
      return deserInfo.defaultValue;
    } else if (deserInfo.typeDeserInfo.type === "array-primitive") {
      deserInfo.defaultValue = [];
      return deserInfo.defaultValue;
    } else if (deserInfo.typeDeserInfo.type === "struct") {
      const defaultValue: Record<string, unknown> = {};
      for (const field of deserInfo.typeDeserInfo.fields) {
        defaultValue[field.name] = this.getFieldDefault(field);
      }
      deserInfo.defaultValue = defaultValue;
      return deserInfo.defaultValue;
    } else if (deserInfo.typeDeserInfo.type === "union") {
      // not sure if the union case should be handled here. At this point the type of the union should be resolved already? 
    }

    throw new Error();
  }
```

Reply:
I know it's a bit verbose, but I wanted to be encapsulate the mutation of the deserInfo in its own function and be clear in the function name that it was `update`ing the variable. Anyway this is definitely confusing, going to make another pass at it and try to consolidate the complex info handling

Reply:
alright updated to remove this function. hopefully it's a bit more clear now.